core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-tables/contributor-admin-stats-table.component.spec.ts

```diff
@@ -301,16 +302,56 @@ describe('Contributor stats component', () => {
         'lastContributedInDays',
         'role',
       ]);
+      expect(component.getOverallItemCount()).toEqual('1+');
     }));
 
-    it('should check for upperlimit for pagination', fakeAsync(() => {
-      component.dataSource = [];
+    it('should check for upperlimit for pagination from items per page', fakeAsync(() => {
       component.statsPageNumber = 1;
-      component.itemsPerPage = 3;
+      component.itemsPerPage = 4;
       fixture.detectChanges();
+      component.allStats = [
+        questionReviewerStats,
+        questionReviewerStats,
+        questionReviewerStats,
+      ];
       expect(component.getUpperLimitValueForPagination()).toEqual(3);
     }));
 
+    it('should check for upperlimit for pagination from number of items', fakeAsync(() => {
+      component.statsPageNumber = 1;
+      component.itemsPerPage = 1;
+      fixture.detectChanges();
+      component.allStats = [
+        questionReviewerStats,
+        questionReviewerStats,
+        questionReviewerStats,
+      ];
+      expect(component.getUpperLimitValueForPagination()).toEqual(2);
+    }));
+
+    it('should show the overall item count from the number of items', fakeAsync(() => {
+      component.statsPageNumber = 1;
+      component.itemsPerPage = 1;
+      fixture.detectChanges();
+      component.allStats = [
+        questionReviewerStats,
+        questionReviewerStats,
+        questionReviewerStats,
+      ];
+      expect(component.getOverallItemCount()).toEqual('3');
```

Comment:
`...toEqual(component.allStats.length.toString())`

Reply:
@StephenYu2018 not sure what is being asked for here ?

Reply:
@stevenjgodfrey The overall item count should be equal to the number of stats in `component.allStats`, correct?

Reply:
Done